account_vendor_bank_account_default/models/res_partner.py

```diff
@@ -0,0 +1,39 @@
+# Copyright 2024 Alberto Martínez <alberto.martinez@sygel.es>
+# License AGPL-3.0 or later (https://www.gnu.org/licenses/agpl).
+
+from odoo import api, fields, models
+
+
+class ResPartner(models.Model):
+    _inherit = "res.partner"
+
+    default_bank_id = fields.Many2one(
+        string="Default Bank Account",
+        comodel_name="res.partner.bank",
+        domain="[('partner_id', '=', commercial_partner_id)]",
+        compute="_compute_default_bank_id",
+        inverse="_inverse_default_bank_id",
+        store=True,
+        readonly=False,
+    )
+    user_default_bank_id = fields.Many2one(
+        comodel_name="res.partner.bank",
+        string="User Default Bank Account",
+    )
+
+    @api.depends("bank_ids", "user_default_bank_id")
+    def _compute_default_bank_id(self):
+        for rec in self:
+            rec.default_bank_id = rec.user_default_bank_id or rec.bank_ids[:1]
```

Comment:
What if you don't have bank_ids? Shouldn't a false be indicated?

Reply:
rec.bank_ids[:1] is False if you don't have bank_ids

Reply:
won't it give a length error? since there are no records and you are trying to access the first one?

Reply:
No, if you access arrays this way no IndexErrors are raised
![image](https://github.com/OCA/bank-payment/assets/45785416/866caa72-acab-49ef-853e-ffe0692b7792)

Reply:
done 😸